internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -348,7 +363,7 @@ func (f *FleetGateway) execute(ctx context.Context) (*fleetapi.CheckinResponse,
 	}
 
 	// get current state
-	state := f.stateFetcher()
+	state, stateCTX := f.stateFetcher.FetchState(ctx)
```

Comment:
```suggestion
	state, stateCtx := f.stateFetcher.FetchState(ctx)
```

nit: we don't capitalize CTX anywhere else in the agent code.